site/src/pages/ManagementSettingsPage/CustomRolesPage/CreateEditRolePageView.tsx

```diff
@@ -0,0 +1,249 @@
+
+  const handleActionCheckChange = async (
+    e: ChangeEvent<HTMLInputElement>,
+    form: ReturnType<typeof useFormik<Role>> & { values: Role },
+  ) => {
+    const { name, checked } = e.currentTarget;
+    const [resource_type, action] = name.split(":");
+
+    const newPermissions = checked
+      ? [
+          ...(checkedActions ?? []),
+          {
+            negate: false,
+            resource_type: resource_type as RBACResource,
+            action: action as RBACAction,
+          },
+        ]
+      : checkedActions?.filter(
+          (p) => p.resource_type !== resource_type || p.action !== action,
+        );
+
+    setCheckActions(newPermissions);
+    await form.setFieldValue("organization_permissions", newPermissions);
+  };
+
+  const resourceActions = showAllResources
+    ? RBACResourceActions
+    : filteredRBACResourceActions;
+
+  return (
+    <>
+      <TableContainer>
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell
+                align="right"
+                sx={{ paddingTop: 0.4, paddingBottom: 0.4 }}
+              >
+                <FormControlLabel
+                  sx={{ marginRight: 1 }}
+                  control={
+                    <Checkbox
+                      size="small"
+                      id="show_all_permissions"
+                      name="show_all_permissions"
+                      checked={showAllResources}
+                      onChange={(e) =>
+                        setShowAllResources(e.currentTarget.checked)
+                      }
+                    />
+                  }
+                  label={
+                    <span style={{ fontSize: 12 }}>Show all permissions</span>
+                  }
+                />
+              </TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {Object.entries(resourceActions).map(([resourceKey, value]) => {
+              return (
+                <TableRow key={resourceKey}>
+                  <TableCell sx={{ paddingLeft: 2 }}>
+                    <li key={resourceKey} css={styles.checkBoxes}>
+                      {resourceKey}
+                      <ul css={styles.checkBoxes}>
+                        {Object.entries(value).map(([actionKey, value]) => {
+                          return (
+                            <li key={actionKey}>
+                              <span css={styles.actionText}>
+                                <Checkbox
+                                  name={`${resourceKey}:${actionKey}`}
+                                  checked={
+                                    checkedActions?.some((p) =>
+                                      ResourceActionComparator(
+                                        p,
+                                        resourceKey,
+                                        actionKey,
+                                      ),
+                                    ) || false
```

Comment:
```suggestion
                                    )
```

it should accept `undefined` as a `false`ish value right?

Reply:
yes, this is probably leftover from an earlier iteration